src/main/java/com/redhat/devtools/intellij/quarkus/lsp4ij/operations/diagnostics/LSPDiagnosticAnnotator.java

```diff
@@ -0,0 +1,145 @@
+        }
+    }
+
+    @Override
+    public @Nullable LSPVirtualFileWrapper doAnnotate(LSPVirtualFileWrapper wrapper) {
+        return wrapper;
+    }
+
+    @Override
+    public void apply(@NotNull PsiFile file, LSPVirtualFileWrapper wrapper, @NotNull AnnotationHolder holder) {
+        if (wrapper.isDiagnosticDirty()) {
+            // LSP Diagnostics has been published for the given file, create for each diagnostic an Intellij Annotation
+            createAnnotations(wrapper, holder);
+            // FIXME : remove this following code, code action request should be called when cursor change or when error annotation is hovered!
+            // Associate for each annotation some quick fixes coming from the code actions request
+            List<Annotation>  annotations = wrapper.getAnnotations();
+            if(annotations != null) {
+                for (Annotation annotation : annotations) {
+                    wrapper.updateQuickFixAt(annotation.getStartOffset());
+                }
+            }
+        } else  {
+            // Update annotation holder with existing annotations
+            // This case comes from:
+            // - when an external component from Intellij Quarkus refresh the validation with DaemonCodeAnalyzer.getInstance(module.getProject()).restart(psiFile);
+            // - when quick fix are updated from LSP code actions.
+         updateAnnotations(wrapper.getAnnotations(), holder);
+        }
+    }
+
+    private void createAnnotations(LSPVirtualFileWrapper wrapper, AnnotationHolder holder) {
+        final List<Diagnostic> diagnostics = wrapper.getAllDiagnostics();
+        VirtualFile virtualFile = wrapper.getFile();
+        Document document = LSPIJUtils.getDocument(virtualFile);
+
+        List<Annotation> annotations = new ArrayList<>();
+        diagnostics.forEach(diagnostic -> {
+            Annotation annotation = createAnnotation(document, diagnostic, holder);
+            if (annotation != null) {
+                if (diagnostic.getTags() != null && diagnostic.getTags().contains(DiagnosticTag.Deprecated)) {
+                    annotation.setHighlightType(ProblemHighlightType.LIKE_DEPRECATED);
+                }
+                annotations.add(annotation);
+            }
+        });
+
+        wrapper.setAnnotations(annotations);
+    }
+
+    private static void updateAnnotations(List<Annotation> existingAnnotations, AnnotationHolder holder) {
+        if (existingAnnotations == null || existingAnnotations.isEmpty()) {
+            return;
+        }
+        // Copy all existing annotations in the given annotation holder.
+        existingAnnotations.forEach(currentAnnotation -> {
+            // Copy range, severity, message information
+            TextRange range = new TextRange(currentAnnotation.getStartOffset() , currentAnnotation.getEndOffset());
+            Annotation newAnnotation = holder.createAnnotation(currentAnnotation.getSeverity(),  range, currentAnnotation.getMessage());

```

Comment:
@angelozerr I tried mirroring these changes in liberty-tools-intellij. I'm getting this Exception whenever an annotation is getting created.
```
Caused by: com.intellij.util.IncorrectOperationException: 'AnnotationHolder.createAnnotation()' method (the call to which was found in class io.openliberty.tools.intellij.lsp4mp.lsp4ij.operations.diagnostics.LSPDiagnosticAnnotator) is slow, non-incremental and thus can cause unexpected behaviour (e.g. annoying blinking), is deprecated and will be removed soon. Please use `newAnnotation(...).create()` instead
```
I am running on IntelliJ 2023.1 with intellij gradle plugin 1.13.3, so possibly it is just deprecated in some of these newer versions.

Reply:
Ok thanks for your feedback. You mean that it doesn't work or it is just a warning message?

Reply:
I hav enot used this create method because it doesn't return the Annotation and I need to store it to update quick fix in async mode. I need to investigate more how to use create method.

Reply:
OK thanks for your feedback. I have investigate how to use the non deprecated api with annotation and I have found a solution. 

I will ping you when I will create the PR

Reply:
@evie-lau please see https://github.com/redhat-developer/intellij-quarkus/pull/819